app/src/test/java/org/simple/clinic/patient/filter/SortByWeightedNamePartsTest.kt

```diff
@@ -0,0 +1,92 @@
+package org.simple.clinic.patient.filter
+
+import com.google.common.truth.Truth
+import junitparams.JUnitParamsRunner
+import junitparams.Parameters
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.simple.clinic.patient.PatientSearchResult.PatientNameAndId
+import java.util.UUID
+
+@RunWith(JUnitParamsRunner::class)
+class SortByWeightedNamePartsTest {
+
+  @Test
+  @Parameters(method = "params for sorting values")
+  fun `it should sort the results based on the overall match of individual parts`(
+      originalList: List<PatientSearchContext>,
+      expected: List<PatientSearchContext>
+  ) {
+    val sorted = originalList.sortedWith(SortByWeightedNameParts())
+
+    Truth.assertThat(sorted).isEqualTo(expected)
```

Comment:
Static import `Truth`?

Reply:
Since I only had a single assertion in this class, I didn't do it. I'll add it in.